### PR TITLE
Add support for MySQLi over SSL, fixes #152

### LINKED ISSUE #152: No SSL with Zend\Db\Adapter\Driver\Mysqli\Connection
There is no way (i didn´t find any) to connect to a mysql server over ssl with the mysqli adapter. 
The method ssl_set() is never called before the connect. it is also impossible to set a flag like MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT with real_connect.

With a little patch it´s possible to use ssl:

  /*\* Config  */
    'db' => array(
        'host'     => '127.0.0.1',
        'dbname'   => 'xxx',
        'username' => 'xxx',
        'password' => 'xxx',
        'ssl'            => true, // Use SSL
        'sslKey'         => ROOT_PATH . '/data/mysql/client-key.pem',  // The path name to the key file
        'sslCert'        => ROOT_PATH . '/data/mysql/client-cert.pem', // The path name to the certificate file
        'sslCaCert'      => ROOT_PATH . '/data/mysql/ca-cert.pem', // The path name to the certificate authority file
        'sslCaPath'      => '', // The pathname to a directory that contains trusted SSL CA certificates in PEM format
        'sslCipher'      => '', // A list of allowable ciphers to use for SSL encryption
    ),

...\vendor\zendframework\zend-db\src\Adapter\Driver\Mysqli\Connection.php in Line 129

```
    /** Patch for SSL */
    $ssl = $findParameterValue(['ssl']);
    $sslKey = $findParameterValue(['sslKey']);
    $sslCert = $findParameterValue(['sslCert']);
    $sslCaCert = $findParameterValue(['sslCaCert']);
    $sslCaPath= $findParameterValue(['sslCaPath']);
    $sslCipher= $findParameterValue(['sslCipher']);
    if($ssl){
        $this->resource->ssl_set($sslKey, $sslCert, $sslCaCert, $sslCaPath, $sslCipher);
    }
```

### PR DESCRIPTION
**Pull request by machek**

### CODE REVIEW

File: test/Adapter/Driver/MySqli/ConnectionIntegrationTest.php [hunk 1]
```diff
@@ -24,16 +24,23 @@ class ConnectionIntegrationTest extends \PHPUnit_Framework_TestCase
      */
     protected function setUp()
     {
+        if(getenv('TESTS_ZEND_DB_ADAPTER_DRIVER_MYSQL')){
```
Comment (froschdesign) [inline test/Adapter/Driver/MySqli/ConnectionIntegrationTest.php]: Skip the test early and no `else` is needed.

File: test/Adapter/Driver/MySqli/ConnectionIntegrationTest.php [hunk 2]
```diff
@@ -24,16 +24,23 @@ class ConnectionIntegrationTest extends \PHPUnit_Framework_TestCase
...
+
+            if (!extension_loaded('mysqli')) {
```
Comment (froschdesign) [inline test/Adapter/Driver/MySqli/ConnectionIntegrationTest.php]: Same here: skip the test early and no nesting is needed.

**Comment by froschdesign** [comment]
@machek 
Btw. please complete your changes and after this you can create a PR. Adding permanently new commits doesn't help.

**Comment by froschdesign** [comment]
If you need the test results from Travis, then you can activate Travis for your own fork.

**Comment by machek** [comment]
@froschdesign sorry about too many commits,Travis behaviour was a bit unexpected, I think would be good to enable integration tests for Mysql, db is available so why not.
Otherwise PR is ready for the review now.
Thanks

**Comment by froschdesign** [comment]
> …I think would be good to enable integration tests for Mysql, db is available so why not.

Sure, but this was not my intention. I mean you can test with Travis at your own fork and prevent permanently new commits at a pull requests.

**Comment by ezimuel** [comment]
@froschdesign what's the status of this PR? I'm working to release zend-db 2.9 and I would like to have this PR included.

**Comment by machek** [comment]
I should mention that coverage/coveralls fails because of

```
    if (!getenv('TESTS_ZEND_DB_ADAPTER_DRIVER_MYSQL')) {
           $this->markTestSkipped('Mysqli integration test disabled');
```
added unit tests are using mysqli adapter because of nature of these changes
but if you feel that some polishing / changes are required please provide feedback, I am happy look into it once more.

**Comment by ezimuel** [comment]
Merged in master and develop. Rebased for master, since this PR was against develop.
@machek thanks for your contribution!

**Pull request closed without merging**